pyam/iiasa.py

```diff
@@ -578,3 +579,60 @@ def read_iiasa(name, default=True, meta=True, creds=None, base_url=_AUTH_URL, **
         Arguments for :meth:`pyam.iiasa.Connection.query`
     """
     return Connection(name, creds, base_url).query(default=default, meta=meta, **kwargs)
+
+
+def lazy_read_iiasa(
+    file, name, default=True, meta=True, creds=None, base_url=_AUTH_URL, **kwargs
+):
+    """Check if the file in a given location is an up-to-date version of an IIASA
+    database. If so, load it. If not, load  data from IIASA scenario explorer and
```

Comment:
I suggest to not mix the terms "database" and "Scenario Explorer". The Scenario Explorer is just a front-end to a database, let us (try to) consistently use the latter.